### ISSUE TITLE
[bug] macos window fixed center

### ISSUE THREAD

**Issue by omaksi** [tauri-apps/tauri#3189]
### Describe the bug

With a clean install on macos, when running `yarn tauri dev` the window is fixed center, always on top, non-movable.

I have tried adding 
```
 "center": false,
 "alwaysOnTop": false
```
to window, but no change.

### Reproduction

`yarn tauri dev` 

### Expected behavior

Window is movable, and not always on top

### Platform and versions

```shell
Operating System - Mac OS, version 11.6.0 X64

Node.js environment
  Node.js - 14.17.1
  @tauri-apps/cli - 1.0.0-beta.10
  @tauri-apps/api - 1.0.0-beta.8

Global packages
  npm - 6.14.13
  yarn - 1.22.17

Rust environment
  rustc - 1.57.0
  cargo - 1.57.0

App directory structure
/node_modules
/public
/src-tauri
/src

App
  tauri.rs - 1.0.0-beta.8
  build-type - bundle
  CSP - default-src blob: data: filesystem: ws: wss: http: https: tauri: 'unsafe-eval' 'unsafe-inline' 'self' img-src: 'self'
  distDir - ../build
  devPath - http://localhost:3000
  framework - React
```

**Comment by amrbashir** [comment]
It works fine for me on Windows, can you provide a minimal repro? 
also Can you try using the `next` branch and see if the issue still exists? 

1. change `tauri` and `tauri-build` versions in `src-tauri/Cargo.toml`.
Edit (@FabianLars): Run `cargo update` after switching to git dependencies!
```
tauri = { git = "https://github.com/tauri-apps/tauri", features = [..] }
tauri-build = { git = "https://github.com/tauri-apps/tauri" }
```
2. install the git version of the cli
```
cargo install tauri-cli --git https://github.com/tauri-apps/tauri
```
3. run the app
```
cargo tauri dev
```

**Comment by omaksi** [comment]
this failed to build with `error[E0658]: use of unstable library feature 'is_symlink'`

```
tauri-utils/src/platform/starting_binary.rs:71:16
.map(std::fs::Metadata::is_symlink)
```

Edit: is a nightly rust version required?

**Comment by FabianLars** [comment]
@omaksi is_symlink is stable since rust@1.57.0 (latest is 1.58.1). So the "fix" is to upgrade your rust version.

Keep in mind that there are huge changes in the next branch since amr posted this (we merged the audit changes and are preparing the first rc), so you may hit more blockers than this.

**Comment by omaksi** [comment]
Ok, updated rust and the build went through, but `yarn tauri dev` returned:

```
process didn't exit successfully: `.../src-tauri/target/debug/build/app-3244ac4e2def1d4d/build-script-build` (exit status: 101)

  --- stdout
  cargo:rerun-if-changed=src/Cargo.toml
  cargo:rerun-if-changed=tauri.conf.json

  --- stderr
  thread 'main' panicked at 'error found during tauri-build: unknown field `readTextFile`, expected one of `scope`, `all`, `readFile`, `writeFile`, `readDir`, `copyFile`, `createDir`, `removeDir`, `removeFile`, `renameFile` at line 1 column 924', .../tauri-9dcc2f9152472c1a/d0f9a03/core/tauri-build/src/lib.rs:103:5
  
```

**Comment by FabianLars** [comment]
Yep, that's one of the pain points. You can't use the npm cli until the next release. So for now it's `cargo tauri dev` (you probably need to _re_-install the cli first so that it includes changes added the last 2 days) -> see amr's step 2 + 3

**Comment by jeff-hykin** [comment]
@FabianLars 
>  is_symlink is stable since rust@1.57.0 (latest is 1.58.1). So the "fix" is to upgrade your rust version.

Are you sure?
<img width="772" alt="Screen Shot 2022-02-28 at 10 16 58 AM" src="https://user-images.githubusercontent.com/17692058/156018407-bc4993d3-e551-4958-9b69-c874900eaa22.png">

**Comment by nothingismagick** [comment]
Is there a reason why you haven't updated rust?

**Comment by lucasfernog** [comment]
Actually `is_symlink` was stabilized in Rust 1.58: https://blog.rust-lang.org/2022/01/13/Rust-1.58.0.html#stabilized-apis

**Comment by FabianLars** [comment]
Ah whoopsie, i was pretty sure it was 1.57 🤷 

Our MSRV is wrong then btw

**Comment by lucasfernog** [comment]
Yeah we forgot to update this. I'll PR it.

**Comment by jeff-hykin** [comment]
> Is there a reason why you haven't updated rust?

Yeah, I use Nixpkg and 1.57.0 is the newest one I see a commit hash for at the moment

**Comment by jeff-hykin** [comment]
I did find a workaround, I just had to dig until I found that installing from the the Dec 28th 2021 commit hash on the `next` branch fixes the issue for me.

(I'm using pinned older versions of Tauri to avoid instability instead of trying to keep up with master)

**Comment by nothingismagick** [comment]
can you share the commit hash please?

**Comment by jeff-hykin** [comment]
Sure!

```shell
# pinned to tauri next-branch ~v1.0.0-beta.7 2021-12-28
cargo install --git https://github.com/tauri-apps/tauri/ --rev efbf236f35a1dd4064c0615b56881e4132a249f4 tauri-cli
```

**Comment by FabianLars** [comment]
This version shouldn't work with tauri@rc, or at least not 100% (not to mention the 3 months of bugfixing...), if you need rc support you can install the cli from this branch: https://github.com/tauri-apps/tauri/pull/3579

**Comment by jeff-hykin** [comment]
Yeah like I said I'm using pinned older versions of Tauri to avoid instability.

```toml
[build-dependencies]
tauri-build = { version = "1.0.0-beta.0", features = [] }

[dependencies]
serde_json = "1.0"
serde = { version = "1.0", features = ["derive"] }
tauri = { version = "= 1.0.0-beta.1", features = ["api-all"] }
tauri-utils = "= 1.0.0-beta.0"
tauri-codegen = "= 1.0.0-beta.0"
tauri-macros = "= 1.0.0-beta.1"
tauri-runtime = "= 0.1.1"
tauri-runtime-wry = "= 0.1.1"
```

I to plan to update everything incrementally, but only after making sure all my test cases pass.

**Comment by nothingismagick** [comment]
We don't support the beta anymore @jeff-hykin

**Comment by jeff-hykin** [comment]
Yup that's fine 👍 @nothingismagick 
I don't need support, just stability.

**Comment by FabianLars** [comment]
i don't see the "stability" here tbh, but who am i to judge 🤷 

Furthermore we should probably stop all that off-topic chatting 😅 

P.S. does anybody know if the actual issue here is resolved on macOS? 😂

**Comment by jeff-hykin** [comment]
When I was trying RC versions I got the error that was mentioned earlier

```
--- stderr
  thread 'main' panicked at 'error found during tauri-build: unknown field `readTextFile`, expected one of `scope`, `all`, `readFile`, `writeFile`, `readDir`, `copyFile`, `createDir`, `removeDir`, `removeFile`, `renameFile` at line 1 column 924', .../tauri-9dcc2f9152472c1a/d0f9a03/core/tauri-build/src/lib.rs:103:5

```

Suggestion I found online was to install Tauri-cli directly from the latest master. That had compiler issues for me.

There was version mismatches + conflicts between tauri-utils, tauri-macros, tauri-runtime, etc (none of them were directly in my cargo file) which is why I had to pin all of them down via guess-and-check to get them to be compatible with each other.

That worked and I can build apps atm, but it's possible the most up to date is still broken on Mac. 


> i don't see the "stability" here tbh, but who am i to judge 🤷

I do think stability is kinda relevant to this issue. The "just update to master / install latest" isn't a good solution for problems like this. Everyone  following that advice is going to have a different experience because master changes so much. It's non-deterministic advice (maybe it'll work maybe it won't; gotta be lucky). That's why, if it worked for one person on date X, I go look up the commit on that date to get their same experience (deterministic/reproducible/stable whatever adj you want to use). I'll take reproducible beta-bugs over being lucky any day, but that's just my preference.

**Comment by nothingismagick** [comment]
@jeff-hykin - please create a new issue or discussion please, because this is really about the question if the center is fixed, or not. 

We NEVER promised that beta would be stable or even work for anyone for any purpose. It was a stepping stone to get us where we are today, and now we are in the RC phase. Yes, software in alpha and beta and RC changes. That is the nature of the beast.

**Comment by jeff-hykin** [comment]
> We NEVER promised that beta would be stable or even work for anyone for any purpose. It was a stepping stone to get us where we are today, and now we are in the RC phase. Yes, software in alpha and beta and RC changes. That is the nature of the beast.

I totally support you guys on that 👍 And I agree 100% with both major breaking changes and the focus of all dev efforts on only the latest version.  Didn't mean for my comment to read like criticism about those choices. First half was supposed to be as a status update on the issue (the outcome of the recommended "install from git") , with the 2nd "stability" half as a separate thought.


> this is really about the question if the center is fixed

My bad @nothingismagick (I found it from searching the `this failed to build with error[E0658]: use of unstable library feature 'is_symlink'` and had assumed it was about that). Once nixpkg's adds rustc 1.58.0, I'll update and retest with latest tauri and open a separate issue for anything I run into.

**Comment by probablykasper** [comment]
I'm not able to reproduce this, so closing for now. Feel free to open a new issue if you experience it again

**Comment by jeff-hykin** [comment]
Encase anyone still hits this, I'm no longer able to reproduce with this setup:

tauri-cli (master as of today) <br> `cargo install --git https://github.com/tauri-apps/tauri/ --rev 6839fde11de39e6eafaee31782825eda1d7f2266 tauri-cli`

Rust
`rustc 1.61.0-nightly (76d770ac2 2022-04-02)`
`cargo 1.61.0-nightly (1ef1e0a12 2022-03-31)`

Cargo.toml
```
[build-dependencies]
tauri-build = { version = "1.0.0-rc.5", features = [] }

[dependencies]
serde_json = "1.0"
serde = { version = "1.0", features = ["derive"] }
tauri = { version = "= 1.0.0-rc.6", features = ["api-all"] }
tauri-utils = "= 1.0.0-rc.4"
tauri-codegen = "= 1.0.0-rc.4"
tauri-macros = "= 1.0.0-rc.4"
tauri-runtime = "= 0.3.4"
tauri-runtime-wry = "= 0.3.5"
```

**Comment by azerpas** [comment]
Yeah I'm getting the same error when running with yarn and rustc 1.58, cargo 1.58.
```
  thread 'main' panicked at 'error found during tauri-build: unknown field `readTextFile`, expected one of `scope`, `all`, `readFile`, `writeFile`, `readDir`, `copyFile`, `createDir`, `removeDir`, `removeFile`, `renameFile` at line 1 column 924', .../tauri-9dcc2f9152472c1a/d0f9a03/core/tauri-build/src/lib.rs:103:5
```

Fixed by running 
```
cargo tauri dev
```

**Comment by lucasfernog** [comment]
Make sure you update every dependency to the RC instead of beta. Use `cargo tauri info` to check it.
